src/gui/window/stats.rs

```diff
@@ -163,8 +171,67 @@ impl GuiWasabiWindow {
                                 ));
                             });
                         }
+                        Statistics::Nps => {
+                            ui.horizontal(|ui| {
+                                ui.monospace("NPS:");
+                                ui.with_layout(
+                                    egui::Layout::right_to_left(egui::Align::Center),
+                                    |ui| {
+                                        self.nps.tick(note_stats.passed_notes.unwrap_or(0) as i64);
+                                        ui.monospace(format!("{}", self.nps.read()));
+                                    },
+                                );
+                            });
+                        }
+                        Statistics::Polyphony => {
+                            if let Some(poly) = stats.polyphony {
+                                ui.horizontal(|ui| {
+                                    ui.monospace("Polyphony:");
+                                    ui.with_layout(
+                                        egui::Layout::right_to_left(egui::Align::Center),
+                                        |ui| {
+                                            ui.monospace(format!("{}", poly));
+                                        },
+                                    );
+                                });
+                            }
+                        }
                     };
                 }
             });
     }
 }
+
+#[derive(Default)]
+pub struct NpsCounter(VecDeque<(Instant, i64)>);
+
+impl NpsCounter {
+    const NPS_WINDOW: f64 = 0.5;
+
+    pub fn tick(&mut self, passed: i64) {
+        self.0.push_back((Instant::now(), passed));
+        while let Some(front) = self.0.front() {
+            if front.0.elapsed().as_secs_f64() > Self::NPS_WINDOW {
+                self.0.pop_front();
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn read(&self) -> u32 {
+        let old = if let Some(front) = self.0.front() {
```

Comment:
```suggestion
        let old = if let Some((front_time, passed)) = self.0.front() {
```
better than using tuples directly